### PR TITLE
Set FIREBASE_AUTH_EMULATOR_HOST in emulators:exec

### PR DESCRIPTION
**Pull request by samtstern**
### Description

This was an oversight ... also I quoted all the IPv6 everywhere now because I always forget.

### Scenarios Tested

```
$ firebase --project=fir-dumpster emulators:exec 'echo $FIREBASE_AUTH_EMULATOR_HOST' --only auth
⚠  Could not find config (firebase.json) so using defaults.
i  emulators: Starting emulators: auth
i  Running script: echo $FIREBASE_AUTH_EMULATOR_HOST
localhost:9099
✔  Script exited successfully (code 0)
i  emulators: Shutting down emulators.
i  auth: Stopping Authentication Emulator
i  hub: Stopping emulator hub
```

### Sample Commands

N/A

### CODE REVIEW

File: src/emulator/commandUtils.ts [hunk 1]
```diff
@@ -335,6 +335,13 @@ async function runScript(script: string, extraEnv: Record<string, string>): Prom
     env[FirestoreEmulator.FIRESTORE_EMULATOR_ENV_ALT] = address;
   }
 
+  const authInstance = EmulatorRegistry.get(Emulators.AUTH);
+  if (authInstance) {
+    const info = authInstance.getInfo();
+    const address = `${info.host}:${info.port}`;
```
Comment (yuchenshi) [inline src/emulator/commandUtils.ts:341]: I really hate to be that person, bit please quote IPv6 addresses

**Review by yuchenshi — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -163,12 +163,14 @@ Use `firebase:deploy --only remoteconfig` to update and publish a project's Fire
...
+The Firebase CLI can use one of four authentication methods listed in descending priority:
+
+- **User Token** - provide an explicit long-lived Firebase user token generated from `firebase login:ci`. Note that these tokens are extremely sensitive long-lived credentials and are not the right option for most cases. Consider using service account authorization instead. The token can be set in one of two ways:
+  - Set the `--token` flag on any command, for example `firebase --token="<token>" projects:list`.
+  - Set the `FIREBASE_TOKEN` environment variable.
+- **Local Login** - run `firebase login` to log in to the CLI directly as yourself. The CLI will cache an authorized user credential on your machine.
+- **Service Account** - set the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to the path of a JSON service account key file.
+- **Application Default Credentials** - if you use the `gcloud` CLI and log in with `gcloud auth application-default login`, the Firebase CLI will use them if none of the above credentials are present.
```
Comment (rosariopfernandes) [inline README.md:173]: Looks like this changes leaked from #2792

Comment (samtstern) [reply README.md:173]: @rosariopfernandes good catch!

File: src/emulator/commandUtils.ts (hunk 1 above)
Comment (samtstern) [reply src/emulator/commandUtils.ts]: @yuchenshi I did it everywhere ... hope that wasn't a bad idea!

**Review by bkendall — APPROVED** [review]

**Review by yuchenshi — APPROVED** [review]

**Pull request merged**